parquet/src/main/java/org/apache/iceberg/parquet/ParquetDictionaryRowGroupFilter.java

```diff
@@ -278,8 +278,27 @@ public Boolean or(Boolean leftResult, Boolean rightResult) {
 
       Set<T> dictionary = dict(id, ref.comparator());
 
-      // ROWS_CANNOT_MATCH if all values of the dictionary are not in the set (the intersection is empty)
-      return Sets.intersection(dictionary, literalSet).isEmpty() ? ROWS_CANNOT_MATCH : ROWS_MIGHT_MATCH;
+      // we need to find out the smaller set to iterate through
+      Set<T> smallerSet;
+      Set<T> biggerSet;
+
+      if (literalSet.size() < dictionary.size()) {
+        smallerSet = literalSet;
+        biggerSet = dictionary;
+      } else {
+        smallerSet = dictionary;
+        biggerSet = literalSet;
+      }
+
+      for (T e : smallerSet) {
```

Comment:
We could rely on `Collections$disjoint` here knowing how it behaves but I'd prefer to have full control here and be sure this logic does not change.

Reply:
Is this equivalent to reversing the order of sets passed to `intersection`?

```java
if (dictionary.size() > literalSet.size()) {
  return Sets.intersection(dictionary, literalSet).isEmpty() ? ROWS_CANNOT_MATCH : ROWS_MIGHT_MATCH;
} else {
  return Sets.intersection(literalSet, dictionary).isEmpty() ? ROWS_CANNOT_MATCH : ROWS_MIGHT_MATCH;
}
```

Reply:
Calling `isEmpty` on `intersection` is equivalent to just calling `Collections$disjoint`. We may rely on the order of args but that seems fragile to me. There is no guarantee the internal implementation won't change.